docs/static/dead-letter-queues.asciidoc

```diff
@@ -0,0 +1,208 @@
+to read from the queue. 
+
+image::static/images/dead_letter_queue.png[Diagram showing pipeline reading from the dead letter queue]
+
+See <<processing-dlq-events>> for more information.
+
+[[configuring-dlq]]
+==== Configuring Logstash to Use Dead Letter Queues
+
+Dead letter queues are disabled by default. To enable dead letter queues, set
+the `dead_letter_queue_enable` option in the `logstash.yml`
+<<logstash-settings-file,settings file>>: 
+
+[source,yaml]
+-------------------------------------------------------------------------------
+dead_letter_queue.enable: true
+-------------------------------------------------------------------------------
+
+Dead letter queues are stored as files in the local directory of the Logstash
+instance. By default, the dead letter queue files are stored in
+`path.data/dead_letter_queue`. Each pipeline has a separate queue. For example,
+the dead letter queue for the `main` pipeline is stored in
+`LOGSTASH_HOME/data/dead_letter_queue/main` by default. The queue files are
+numbered sequentially: `1.log`, `2.log`, and so on.
+
+You can set `path.dead_letter_queue` in the `logstash.yml` file to
+specify a different path for the files:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+path.dead_letter_queue: "path/to/data/dead_letter_queue"
+-------------------------------------------------------------------------------
+
+===== File Rotation
+
+Dead letter queues have a built-in file rotation policy that manages the file
+size of the queue. When the file size reaches a preconfigured threshold,  a new
+file is created automatically. The size limit of the dead letter queue is
+constrained only by the amount of space that you have available on disk.
+
+NOTE: Dead letter queues retain all the events that are written to them.
+Currently, you cannot configure the size of the queue or the size of the files
+that are used to store the queue. 
+
+[[processing-dlq-events]]
+==== Processing Events in the Dead Letter Queue
+
+When you are ready to process events in the dead letter queue, you create a
+pipeline that uses the `dead_letter_queue` input plugin to read from the dead
+letter queue. The pipeline configuration that you use depends, of course, on
+what you need to do. For example, if the dead letter queue contains events that
+resulted from a mapping error in Elasticsearch, you can create a pipeline that
+reads the "dead" events, removes the field that caused the mapping issue, and
+re-indexes the clean events into Elasticsearch. 
+
+The following example shows a simple pipeline that reads events from the dead
+letter queue and writes the events to standard output: 
+
+[source,yaml]
+--------------------------------------------------------------------------------
+input {
+  dead_letter_queue {
+    path => "/path/to/data/dead_letter_queue" <1>
+    commit_offsets => true <2>
+    pipeline_id => "main" <3>
+  }
+}
+
+output {
+  stdout {
+    codec => rubydebug
```

Comment:
Can we use `stdout { codec => rubydebug { metadata => true } }` in this example. If you add `metadata => true`, LS will emit the metadata information which is useful in the DLQ case.